### PR TITLE
chore(): Remove over-protective cloneDeep from fromObject

### PR DESCRIPTION
**Pull request by asturur**
## Description

We need to protect from unwanted references when restoring from data.
Today we deeply clone every object, and that is expensive ( measured on an app with 91 canvases  it went from 480 to around 420 ms ) and overly protective.

In the case of Object._fromObject `options` ( renamed in `serializedOptions` ) is passing through the enliving process that already a shallow clone

Other deep level cloning are left to the single objects duty or single properties ( gradients, pattern, array ) but they need to be carefully inspected

This PR is a wip and will add more checks on data cloning later

### CODE REVIEW

File: src/gradient/Gradient.ts [hunk 1]
```diff
@@ -112,14 +112,19 @@ export class Gradient<
...
     this.coords = {
       ...(this.type === 'radial' ? radialDefaultCoords : linearDefaultCoords),
       ...coords,
     } as GradientCoords<T>;
-    this.colorStops = colorStops.slice();
+    this.colorStops = colorStops.map((colorStop) => ({
+      color: colorStop.color,
+      offset: colorStop.offset,
+      opacity: colorStop.opacity,
+    }));
```
Comment (asturur) [inline src/gradient/Gradient.ts:127]: Gradient should have a fromObject like Pattern

File: src/shapes/Object/Object.ts [hunk 2]
```diff
@@ -1564,20 +1563,22 @@ export class FabricObject<
...
   static _fromObject<S extends FabricObject>(
-    { type, ...object }: Record<string, unknown>,
+    { type, ...serializedObjectOptions }: Record<string, unknown>,
     { extraParam, ...options }: Abortable & { extraParam?: string } = {}
   ): Promise<S> {
-    return enlivenObjectEnlivables<any>(cloneDeep(object), options).then(
-      (enlivedMap) => {
-        const allOptions = { ...options, ...enlivedMap };
+    return enlivenObjectEnlivables<any>(serializedObjectOptions, options).then(
+      (enlivedObjectOptions) => {
```
Comment (asturur) [inline src/shapes/Object/Object.ts:1570]: enlivedObjectOptions is a shallow clone of object so we can delete or reassign properties freely.

File: src/shapes/Object/Object.ts [hunk 3]
```diff
@@ -1564,20 +1563,22 @@ export class FabricObject<
    * @returns {Promise<FabricObject>}
    */
   static _fromObject<S extends FabricObject>(
-    { type, ...object }: Record<string, unknown>,
+    { type, ...serializedObjectOptions }: Record<string, unknown>,
     { extraParam, ...options }: Abortable & { extraParam?: string } = {}
   ): Promise<S> {
-    return enlivenObjectEnlivables<any>(cloneDeep(object), options).then(
-      (enlivedMap) => {
-        const allOptions = { ...options, ...enlivedMap };
```
Comment (asturur) [inline src/shapes/Object/Object.ts:1572]: the only other valuable here should be signal. not worth a full spread.

File: src/gradient/Gradient.ts (hunk 1 above)
Comment (ShaMan123) [reply src/gradient/Gradient.ts:127]: #9144

**Pull request merged**